app/controllers/api/v1/authorization_definitions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class API::V1::AuthorizationDefinitionsController < API::V1Controller
+  before_action -> { doorkeeper_authorize! :read_authorizations }, only: %i[index show]
+  before_action :set_authorization_definition, only: %i[show]
+
+  def index
+    authorization_definitions = AuthorizationDefinition.all.select do |definition|
+      current_user_authorization_request_types.include?(definition.authorization_request_class.to_s)
+    end
```

Comment:
```suggestion
    authorization_definitions = AuthorizationDefinition.where(authorization_request_class: current_user_authorization_request_types)
```

? je ne suis pas sûr que cela marche mais niveau interface ça serait quand même mieux.. quitte à itérer sur le `StaticApplicationRecord`

Reply:
ça me fait chier ici par ce que `authorization_request_class` c'est pas une string dans AuthorizationDefinition, c'est vraiment la classe. Et j'ai pas envie de const_get les string d'input utilisateur pour en faire des classes